devtools/maven/src/main/resources/create-extension-templates/deployment-pom.xml

```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>[=groupId]</groupId>
+        <artifactId>[=artifactId]-parent</artifactId>
+        <version>[=version]</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>[=artifactId]-deployment</artifactId>
+[#if nameBase?? ]    <name>[=namePrefix][=nameBase][=nameSegmentDelimiter]Deployment</name>
+[/#if]
+
+    <dependencies>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-core-deployment</artifactId>
+[#if !assumeManaged ]            <version>[=quarkusVersion]</version>
+[/#if]
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
```

Comment:
This assumes the plugin management is inherited from the parent, correct? This either has to be documented or `<version>[=quarkusVersion]</version>` should be added here and in other templates.

Reply:
> This assumes the plugin management is inherited from the parent, correct? 

Yes, I assume sane projects manage their plugins somewhere up in their module hierarchy.

> This either has to be documented 

Documented where? In the Mojo JavaDoc?

> or `<version>[=quarkusVersion]</version>` should be added here and in other templates.

Well, `<version>[=quarkusVersion]</version>` is not correct for sure. I think if we want to be perfect, we should have a dedicated `compilerPluginVersion` Mojo param. But personally, I'd say this is not worth overengineering at the current stage.